sync/src/main/java/tech/pegasys/teku/sync/multipeer/BatchImporter.java

```diff
@@ -46,19 +50,23 @@ public BatchImporter(final BlockImporter blockImporter, final AsyncRunner asyncR
    * @return a future reporting the result of the import
    */
   public SafeFuture<BatchImportResult> importBatch(final Batch batch) {
-    // Copy the blocks as we're going to use them from off the event thread.
+    // Copy the data from batch as we're going to use them from off the event thread.
     final List<SignedBeaconBlock> blocks = new ArrayList<>(batch.getBlocks());
+    final Map<SignedBeaconBlock, SyncSource> blockToSource =
+        blocks.stream().collect(Collectors.toUnmodifiableMap(b -> b, batch::getBlockSource));
```

Comment:
All blocks in the batch will have come from the same source so we can probably skip the map here.  Otherwise all the other checks and penalties don't work out anyway.

Reply:
Makes sense - will simplify.